### PR TITLE
Feature/target soft delete

### PR DESCRIPTION
**Pull request by agwozdowski**

### CODE REVIEW

File: app/admin/target_scopes.rb [hunk 1]
```diff
@@ -1,7 +1,46 @@
...
+    f.inputs do
+      f.input :name
+    end
+
+    f.actions
+  end
+
+  controller do
+    def destroy
+      destroy_command = ::Command::Destroy::TargetScope.new(resource)
```
Comment (kowal) [inline app/admin/target_scopes.rb]: Maybe it's worth to extract this whole action to some module (parametrized with resource class). It looks like it won't change much anymore.

Comment (agwozdowski) [reply app/admin/target_scopes.rb]: @kowal added DiscardableController. Also moved DiscardableModel to concerns. Take a look please

File: app/controllers/concerns/discardable_controller.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+module DiscardableController
+  # Override default Active Admin destroy method using Discardable Model.
+  # It sets discarded_at for entity and for its dependent resources.
+  # Redirect to resource list with proper message
+  def destroy
+    entity_name = ActiveSupport::Inflector.singularize(controller_name).camelize
+    entity_destroy_class = "::Command::Destroy::#{entity_name}".constantize
```
Comment (kowal) [inline app/controllers/concerns/discardable_controller.rb]: can we catch possible `NameError`s here and re-throw ex with more context?

Comment (agwozdowski) [reply app/controllers/concerns/discardable_controller.rb]: @kowal done

**Pull request merged**